token-client/src/main/java/com/sap/cloud/security/xsuaa/client/AbstractOAuth2TokenService.java

```diff
@@ -187,6 +187,31 @@ public OAuth2TokenResponse retrieveAccessTokenViaJwtBearerTokenGrant(URI tokenEn
 		return getOAuth2TokenResponse(tokenEndpoint, headers, parameters, subdomain, disableCacheForRequest);
 	}
 
+	public OAuth2TokenResponse retrieveAccessTokenViaJwtBearerTokenGrant(URI tokenEndpoint,
+			ClientCredentials clientCredentials, @Nonnull String token,
+			@Nullable Map<String, String> optionalParameters, boolean disableCacheForRequest,
+			@Nonnull String xZidHeader)
+			throws OAuth2ServiceException {
+		assertNotNull(tokenEndpoint, "tokenEndpoint is required");
+		assertNotNull(clientCredentials, "clientCredentials are required");
+		assertNotNull(token, "token is required");
+		assertNotNull(xZidHeader, "X-zid header is required");
+
+		Map<String, String> parameters = new RequestParameterBuilder()
+				.withGrantType(GRANT_TYPE_JWT_BEARER)
+				.withClientCredentials(clientCredentials)
+				.withToken(token)
+				.withOptionalParameters(optionalParameters)
+				.buildAsMap();
+
+		HttpHeaders headers = HttpHeadersFactory.createWithXzidHeader(xZidHeader);
```

Comment:
i would suggest sth like:
HttpHeaders headers = HttpHeadersFactory.createWithoutAuthorizationHeader().with(HttpHeaders.X_ZID, xZidHeader);

Reply:
to chain the methods I would need to convert the methods to instance methods, or is there another way for method chaining in static context:question:

Reply:
Hi @liga-oz 
The "with" method will become an instance method of HttpHeaders, which then returns itself as instance of type HttpHeaders)...